app/templates/components/insights-tabs.hbs

```diff
@@ -0,0 +1,7 @@
+{{#each tabs as |tab|}}
+  <li class="insights-tab {{if (eq tab.slug selectedTab) 'active'}}">
+    {{#link-to (query-params timeInterval=tab.slug) }}

```

Comment:
I'm not sure if it would work in this case, but worth mentioning that when the query parameter is active on URL, the link-to helper adds active class to the link automatically. Could it be used here instead of manually setting active on li?

Reply:
Oooh nice, I'll test it out and see if it would work in this case.

Reply:
It works, but breaks the component test that checks that active states are set correctly. Do you know of a way to set up or pass in query params for such a test? Or maybe it would make sense to move that test out to the acceptance test for the route? I'm leaning toward moving it out to the acceptance test, wdyt?

Reply:
Yeah I really like having the query-params helper automatically handle stuff, so I think that's what I'm gonna go with.

Reply:
Query parameters are only available on route and constructor level. So you could get the `timeInterval` property value on constructor and pass it to the component (in case if it needs to know the state). In test you could assign this property directly:

```
export default Controller.extend({
  queryParams: ['timeInterval'],
  timeInterval: DEFAULT_INSIGHTS_INTERVAL
});
```
```
export default Component.extend({
  tagName: 'ul',
  classNames: ['travistab-nav travistab-nav--insights insights-tabs'],

   tabs: INSIGHTS_TABS,
   activeTab: DEFAULT_INSIGHTS_INTERVAL
});
```
```
module('Integration | Component | insights-tabs', function (hooks) {
  setupRenderingTest(hooks);

   test('display', async function (assert) {
    await render(hbs`{{insights-tabs activeTab='month' }}`);

     assert.dom('.insights-tabs').exists();
    assert.dom('.insights-tabs .insights-tab').exists({ count: Object.values(INSIGHTS_INTERVALS).length });
    assert.dom('.insights-tabs .insights-tab .active').hasText('month');
  });
});
```

Although, I have noticed that you got rid of this logic entire, which works as well :)

Reply:
Yeah that's what I suspected. Thanks!